Back-end/Service/UserService.cs

```diff
@@ -0,0 +1,180 @@
+﻿using DENMAP_SERVER.Entity;
+using DENMAP_SERVER.Repository;
+using MySql.Data.MySqlClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DENMAP_SERVER.Service
+{
+    internal class UserService
+    {
+        string connectionString = "Server=34.116.253.154;Port=3306;Database=chat_database;Uid=app_user;Pwd=&X9fT#7vYqZ$4LpR;";
+
+        private UserRepository userRepository = new UserRepository();
+
+        public int RegisterUser(string username, string password, byte[] image, string description)
+        {
+            int id = 0;
+            User user = null;
+
+            using (MySqlConnection connection = new MySqlConnection(connectionString))
+            {
+                connection.Open();
+                try
+                {
+                    user = userRepository.getUserByName(connection, username);
+
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception("Database error.\nError:" + ex.Message);
+                }
+
+                if (user != null)
+                {
+                    throw new Exception("User with name: " + username + " already exists");
+                }
+
+                try
+                {
+                    id = userRepository.addUser(connection, username, password, image, description);
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception("Database error.\nError:" + ex.Message);
+                }
+            }
+            return id;
+        }
+
+        public User getUserByName(string name)
+        {
+            User user = null;
+            using (MySqlConnection connection = new MySqlConnection(connectionString))
+            {
+                connection.Open();
+                try
+                {
+                    user = userRepository.getUserByName(connection, name);
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception("Database error.\nError:" + ex.Message);
+                }
+            }
+
+            if (user == null)
+            {
+                throw new Exception("User with name: " + name + " not found");
+            }
+
+
+            return user;
+        }
+
+        public int loginUser(string name, string password)
+        {
+            User user = null;
+
+            using (MySqlConnection connection = new MySqlConnection(connectionString))
+            {
+                connection.Open();
+                System.Console.WriteLine("User service, starting login");
```

Comment:
Create real logger to this project that will log also to file(in the future commits)